File: frontend/src/objects/hack/ClaimedHackAttempt.ts
import type { HackId } from '@/objects/hack/HackId'
import type { ClaimedSubmission } from '@/objects/submission/ClaimedSubmission'
import type { JudgeResult } from '@/objects/submission/JudgeResult'
import type { Username } from '@/objects/user/Username'

/** Hack worker 领取到的 hack attempt；镜像后端内部 ClaimedHackAttempt。 */
export type ClaimedHackAttempt = {
  hackId: HackId
  targetSubmission: ClaimedSubmission
  authorUsername: Username
  subtaskIndex: number
  input: string
  strategyProviderSource: string | null
  oldResult: JudgeResult
}

File: backend/src/main/scala/domains/hack/objects/internal/ClaimedHackAttempt.scala
package domains.hack.objects.internal

import domains.hack.objects.HackId
import domains.submission.objects.internal.ClaimedSubmission
import domains.user.objects.Username
import judgeprotocol.objects.response.JudgeResult

/** hack 领取任务载荷，由 ClaimNextHackAttempt 内部 API 返回，ClaimJudgeTask 用它组装 judge-protocol 的 HackTask。 */
final case class ClaimedHackAttempt(
  hackId: HackId,
  targetSubmission: ClaimedSubmission,
  authorUsername: Username,
  subtaskIndex: Int,
  input: String,
  strategyProviderSource: Option[String],
  oldResult: JudgeResult
)
